### ISSUE TITLE
"Tried to enable notifications handler that was already enabled"

### ISSUE THREAD

**Issue by tomaka** [paritytech/substrate#5201]
Some warnings have been noticed on Polkadot master.

```
2020-03-07 19:24:12 Imported #1351071 (0x2a59…4c87)
2020-03-07 19:24:20 Discovered new external address for our node: /ip4/10.156.1.213/tcp/30333/p2p/QmSz5s5cmK2x2A1BKMyWvtTX2Sy6LzmPZN1AJPzMvWfiEK
2020-03-07 19:24:20 Imported #1351071 (0x9113…db21)
2020-03-07 19:24:20 Imported #1351071 (0xec8c…07b5)
2020-03-07 19:24:21 Tried to enable notifications handler that was already enabled
2020-03-07 19:24:21 Tried to enable notifications handler that was already enabled
2020-03-07 19:24:21 Tried to enable notifications handler that was already enabled
2020-03-07 19:24:21 Imported #1351072 (0x62d8…9747)
2020-03-07 19:24:21 Notifications handler in a poisoned state
2020-03-07 19:24:21 Notifications handler in a poisoned state
2020-03-07 19:24:21 Notifications handler in a poisoned state
2020-03-07 19:24:21 Imported #1351072 (0xe3b5…5796)
```

```
2020-03-08 08:45:18 Tried to send handshake twice
2020-03-08 08:45:18 Tried to send handshake twice
2020-03-08 08:45:18 Tried to send handshake twice
```

Having the `sub-libp2p` logs would be extremely useful, however I tried running a node with `--in-peers 200` for 12 hours and couldn't reproduce.

Messages are printed three times because Polkadot has three active notification protocols, but this probably concerns only one connection.

**Comment by tomaka** [comment]
The "Tried to send handshake twice" message is fixed by #5413, but the other messages still need to be debugged.

**Comment by tomaka** [comment]
I couldn't find this in the logs of the Parity nodes in the past 30 days, so I'm going to close as "fixed by accident".
